server.go

```diff
@@ -400,7 +400,9 @@ func (s *Server) controlHandler(w http.ResponseWriter, r *http.Request) (ctErr e
 		return fmt.Errorf("error writing response: %s", err)
 	}
 
-	conn.SetDeadline(time.Time{})
+	if err := conn.SetDeadline(time.Time{}); err != nil {
```

Comment:
It can [when connection is not ok](https://github.com/golang/go/blob/master/src/net/net.go#L238), this is kind of check you have in all sorts of net.conn functions.